lib/codegen/meta/src/srcgen.rs

```diff
@@ -45,6 +45,18 @@ impl Formatter {
         ret
     }
 
+    pub fn indent_with<T>(
```

Comment:
I consider having something like this too, but I never implemented it, because I thought it made it actually slightly harder to follow the generated code. At least with explicit `line()` calls before and after, we don't have to jump around to see what's after the block we just indented. If that makes sense, I'd just advise to use plain `line()` calls before and after instead.